code/game/world.dm

```diff
@@ -133,19 +133,22 @@ GLOBAL_VAR(restart_counter)
 	if (bannedsourceaddrs[addr])
 		return
 
-	if(length(T) >= MAX_TOPIC_LEN)
-		log_admin_private("[addr] banned from topic calls for a round for too long status message")
-		bannedsourceaddrs[addr] = TOPIC_BANNED
-		return
-
-	if(lasttimeaddr[addr])
-		var/lasttime = lasttimeaddr[addr]
-		if(world.time < (lasttime + 2 SECONDS))
-			log_admin_private("[addr] banned from topic calls for a round for too frequent messages")
+	var/list/filtering_whitelist = CONFIG_GET(keyed_list/topic_filtering_whitelist)
+	var/host = splittext_char(addr, ":")
```

Comment:
Won't `addr` here always be ASCII? If so, you can just use `splittext` instead

Also, where are you comparing the address?

Reply:
`if(!filtering_whitelist[host[1]])`

`filtering_whitelist` is a map